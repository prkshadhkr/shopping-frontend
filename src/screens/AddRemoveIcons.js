import React from 'react';
import { useDispatch } from 'react-redux';
import { addItem, removeItem } from '../action/products';

const AddRemoveIcons = ({ id }) =>{
  const dispatch = useDispatch();
  const add = () => dispatch(addItem(id));
  const remove = () => dispatch(removeItem(id));

  return (
    <div>
      <div>
        <i className="fas fa-plus-square fa-1x text-success" onClick={add} />
        <i className="fas fa-minus-square fa-1x text-danger" onClick={remove} />    
      </div>
    </div>
  )

}
export default AddRemoveIcons;