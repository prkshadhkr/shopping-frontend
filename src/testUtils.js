import React from  'react';
import UserContext from './user/UserContext';

/** following is for mock store */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(axios);
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};
const makeMockStore = (state = {}) => {
  return mockStore ({
    ...initialState,
    ...state
  })
}

/** create a demouser */
const demoUser = {
  username: "testing",
  first_name: "testing",
  last_name: "testing",
  email: "testing@gmail.com",
  password: "testing",
  is_admin: false
};

/**
 * user is destructured from userContext in Profilejs as
 * const { user, setUser } = useContext(UserContext);
 */
const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
)

export { mock, makeMockStore, UserProvider };