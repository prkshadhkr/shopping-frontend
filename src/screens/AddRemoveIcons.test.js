import React from 'react';
import { render } from '@testing-library/react';
import AddRemoveIcons from './AddRemoveIcons';
import { MemoryRouter } from 'react-router';
import { Provider } from "react-redux";
import { makeMockStore } from '../testUtils';

const store = makeMockStore();

it ('renders without crashing', function() {
  render(
    <MemoryRouter>
      <Provider store={store}>
        <AddRemoveIcons />
      </Provider>
    </MemoryRouter>
  )
});

it ('matches snapshot', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <Provider store={store}>
        <AddRemoveIcons />
      </Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})