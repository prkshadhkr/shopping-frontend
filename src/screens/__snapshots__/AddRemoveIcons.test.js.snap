// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <i
        class="fas fa-plus-square fa-1x text-success"
        style="margin-right: 5px;"
      />
      <i
        class="fas fa-minus-square fa-1x text-danger"
      />
    </div>
  </div>
</DocumentFragment>
`;
